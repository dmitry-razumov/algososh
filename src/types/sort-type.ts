export enum SortType {
  Selection = "selection",
  Bubble = "bubble"
}