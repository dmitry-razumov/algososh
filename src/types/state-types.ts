import { ElementStates } from "./element-states"

export type TElement = {
  value: string,
  state: ElementStates,
}

export type TArrayStep<T> = {
  array: T[]
}