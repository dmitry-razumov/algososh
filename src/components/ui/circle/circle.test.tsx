import { render } from "@testing-library/react"
import { Circle } from "./circle"
import { ElementStates } from "../../../types/element-states"

describe('Circle component tests', () => {
  test('Should render without letter', () => {
    render(<Circle />)
    expect(document.body).toMatchSnapshot();
  })

  test('Should render with letter', () => {
    render(<Circle letter="XY"/>)
    expect(document.body).toMatchSnapshot();
  })

  test('Should render with head', () => {
    render(<Circle head="head"/>)
    expect(document.body).toMatchSnapshot();
  })

  test('Should render with react element in head', () => {
    render(<Circle head={<Circle />}/>)
    expect(document.body).toMatchSnapshot();
  })

  test('Should render with tail', () => {
    render(<Circle tail="tail"/>)
    expect(document.body).toMatchSnapshot();
  })

  test('Should render with react element in tail', () => {
    render(<Circle tail={<Circle />}/>)
    expect(document.body).toMatchSnapshot();
  })

  test('Should render with index', () => {
    render(<Circle index={1}/>)
    expect(document.body).toMatchSnapshot();
  })

  test('Should render with prop isSmall={true}', () => {
    render(<Circle isSmall={true}/>)
    expect(document.body).toMatchSnapshot();
  })

  test('Should render with state Default', () => {
    render(<Circle state={ElementStates.Default}/>)
    expect(document.body).toMatchSnapshot();
  })

  test('Should render with state Changing', () => {
    render(<Circle state={ElementStates.Changing}/>)
    expect(document.body).toMatchSnapshot();
  })
  
  test('Should render with state Modified', () => {
    render(<Circle state={ElementStates.Modified}/>)
    expect(document.body).toMatchSnapshot();
  })
})