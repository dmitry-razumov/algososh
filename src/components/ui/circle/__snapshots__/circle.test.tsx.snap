// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component tests Should render with head 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head string"
      >
        head
      </div>
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</body>
`;

exports[`Circle component tests Should render with index 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      >
        1
      </p>
      <div
        class="text text_type_input text_color_input mt-4 absolute tail60 element"
      />
    </div>
  </div>
</body>
`;

exports[`Circle component tests Should render with letter 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        >
          XY
        </p>
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</body>
`;

exports[`Circle component tests Should render with prop isSmall={true} 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle  small default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</body>
`;

exports[`Circle component tests Should render with react element in head 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      >
        <div
          class="content "
        >
          <div
            class="text text_type_input text_color_input mb-4 absolute head element"
          />
          <div
            class="circle   default"
          >
            <p
              class="text text_type_circle text_color_input letter"
            />
          </div>
          <p
            class="text text_type_input text_color_input mt-4 absolute index"
          />
          <div
            class="text text_type_input text_color_input mt-4 absolute tail30 element"
          />
        </div>
      </div>
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</body>
`;

exports[`Circle component tests Should render with react element in tail 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      >
        <div
          class="content "
        >
          <div
            class="text text_type_input text_color_input mb-4 absolute head element"
          />
          <div
            class="circle   default"
          >
            <p
              class="text text_type_circle text_color_input letter"
            />
          </div>
          <p
            class="text text_type_input text_color_input mt-4 absolute index"
          />
          <div
            class="text text_type_input text_color_input mt-4 absolute tail30 element"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Circle component tests Should render with state Changing 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   changing"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</body>
`;

exports[`Circle component tests Should render with state Default 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</body>
`;

exports[`Circle component tests Should render with state Modified 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   modified"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</body>
`;

exports[`Circle component tests Should render with tail 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 string"
      >
        tail
      </div>
    </div>
  </div>
</body>
`;

exports[`Circle component tests Should render without letter 1`] = `
<body>
  <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</body>
`;
